mealie/routes/auth/auth.py

```diff
@@ -48,12 +49,15 @@ def respond(cls, token: str, token_type: str = "bearer") -> dict:
 
 
 @public_router.post("/token")
-def get_token(data: CustomOAuth2Form = Depends(), session: Session = Depends(generate_session)):
+def get_token(request: Request, data: CustomOAuth2Form = Depends(), session: Session = Depends(generate_session)):
+    settings = get_app_settings()
+
+    jwt_assertion = request.headers.get(settings.JWT_AUTH_HEADER_NAME, None)
     email = data.username
     password = data.password
 
     try:
-        user = authenticate_user(session, email, password)  # type: ignore
+        user = authenticate_user(session, email, password, jwt_assertion)  # type: ignore
```

Comment:
If present in the configured header, the JWT assertion will be passed to the `authenticate_user` function.